traversal/planner/GraphPlanner.java

```diff
@@ -218,134 +162,101 @@ private PlannerVertex.Type typeVertex(StructureVertex.Type structureVertex) {
-        assert !Double.isNaN(totalCostLastRecorded);
-        assert !Double.isNaN(costPrevious);
-        assert !Double.isNaN(costNext);
-        assert costPrevious > 0 && totalCostLastRecorded > 0;
+    @Override
+    public GraphPlanner asGraph() {
+        return this;
+    }
 
-        totalCostNext += costNext;
-        assert !Double.isNaN(totalCostNext);
+    public Collection<PlannerVertex<?>> vertices() {
+        return vertices.values();
+    }
 
-        if (costNext / costPrevious >= OBJECTIVE_VARIABLE_COST_MAX_CHANGE &&
-                costNext / totalCostLastRecorded >= OBJECTIVE_VARIABLE_TO_PLANNER_COST_MIN_CHANGE) {
-            setOutOfDate();
-        }
+    public Set<PlannerEdge<?, ?>> edges() {
+        return edges;
+    }
+
+    void setOutOfDate() {
+        this.isUpToDate = false;
     }
 
-    private void setBranchingFactor(GraphManager graph) {
-        // TODO: We can refine the branching factor by not strictly considering entities being the only divisor
-        double entities = graph.data().stats().thingVertexTransitiveCount(graph.schema().rootEntityType());
-        double roles = graph.data().stats().thingVertexTransitiveCount(graph.schema().rootRoleType());
-        if (roles == 0) roles += 1;
-        if (entities > 0) branchingFactor = roles / entities;
-        assert !Double.isNaN(branchingFactor);
+    private boolean isUpToDate() {
+        return isUpToDate;
     }
 
-    private void setCostExponentUnit(GraphManager graph) {
-        double expUnit, expMaxInc, expMax;
-        expUnit = (OBJECTIVE_COEFFICIENT_MAX_EXPONENT_DEFAULT - 1) / edges.size();
-        expUnit = Math.min(expUnit, 1.0);
+    private boolean isPlanned() {
+        return optimiser.isFeasible() || optimiser.isOptimal();
+    }
 
-        expMaxInc = expUnit * edges.size();
-        expMax = 1 + expMaxInc;
-        long things = graph.data().stats().thingVertexTransitiveCount(graph.schema().rootThingType());
-        double maxCoefficient = Math.pow(things, expMax);
-        if (Double.isNaN(maxCoefficient) || Double.isInfinite(maxCoefficient) || maxCoefficient > Long.MAX_VALUE) {
-            expMax = Math.log(Long.MAX_VALUE) / Math.log(things);
-            expMaxInc = expMax - 1;
-        }
-        assert !Double.isNaN(expMaxInc) && expMaxInc > 0;
-        costExponentUnit = expMaxInc / edges.size();
+    @Override
+    public boolean isOptimal() {
+        return optimiser.isOptimal();
     }
 
-    private void computeTotalCostNext(GraphManager graph) {
-        vertices.values().forEach(v -> v.updateObjective(graph));
-        edges.forEach(e -> e.updateObjective(graph));
+    private boolean isError() {
+        return optimiser.isError();
     }
 
-    private void setInitialValues() {
-        new Initialiser().execute();
+    Optimiser optimiser() {
+        return optimiser;
     }
 
-    void mayOptimise(GraphManager graph, boolean singleUse) {
+    void mayOptimise(GraphManager graphMgr, boolean singleUse) {
```

Comment:
`GraphManager graph` -> `GraphManager graphMgr` everywhere, the planner seemed to be the only place where that naming convention was violated.